pilot/pkg/bootstrap/sidecarinjector.go

```diff
@@ -91,15 +92,17 @@ func (s *Server) initSidecarInjector(args *PilotArgs) (*inject.Webhook, error) {
 	// operator or CI/CD
 	if features.InjectionWebhookConfigName != "" {
 		s.addStartFunc(func(stop <-chan struct{}) error {
-			// No leader election - different istiod revisions will patch their own cert.
-			// update webhook configuration by watching the cabundle
-			patcher, err := webhooks.NewWebhookCertPatcher(s.kubeClient, args.Revision, webhookName, s.istiodCertBundleWatcher)
-			if err != nil {
-				log.Errorf("failed to create webhook cert patcher: %v", err)
-				return nil
-			}
-
-			patcher.Run(stop)
+			go leaderelection.NewPerRevisionLeaderElection(args.Namespace, args.PodName, args.Revision, leaderelection.MutatingWebhookController, s.kubeClient.Kube()).
```

Comment:
addTerminatingStartFunc is slightly preferred, will give us a better chance at releasing the lock on shutdown (so another instance can immediately start).

Reply:
Cool did not know about this, done.